### PR TITLE
harden edge cases on ufm drift

### PR DESCRIPTION
**Pull request by susanhooks**
## Validation

- [x] Standard CI passes.
- [x] Kind integration passes, or this PR explains why it was not run.

The kind integration test is manual due to taking ~30 min to complete. When the PR is ready for review,
run Actions -> Kind Integration -> Run workflow against the copy-pr-bot generated
`pull-request/<PR_NUMBER>` branch. Use the default `test_path` for the full suite,
or narrow it only while debugging.

Passing Kind Integration run:

https://github.com/NVIDIA/nv-config-manager/actions/runs/28391160738
## Checklist

- [x] I am familiar with the contributing guidelines in `CONTRIBUTING.md`.
- [x] Commits are signed off for DCO compliance.
- [x] New or existing tests cover these changes, or the PR explains why tests are not needed.
- [x] Documentation is updated for user-facing behavior changes.
- [x] Generated artifacts are updated when applicable, such as OpenAPI specs,
      docs screenshots, or Helm/rendered outputs.

## Summary by CodeRabbit

* **New Features**
  * Update/delete flows now detect UFM-only PKey members and surface them in the approval diff, using UFM membership state during reconciliation.
  * Deletion cleanup now propagates UFM “partition empty” status to decide whether to remove Nautobot PKey/overlay data.
  * PKey updates preserve UFM’s existing `ip_over_ib` during reconciliation.
* **Bug Fixes**
  * Treats UFM “PKey not found” (404) as non-existent instead of failing, and improves reconciliation logic for remaining/untracked members.
* **Tests**
  * Added/updated coverage for UFM-only blocking cleanup, empty-but-not-empty handling, and backward-compatible defaults.

### CODE REVIEW

File: src/tests/temporal/ngc/activities/test_ib_pkey_update.py [hunk 1]
```diff
@@ -164,15 +164,55 @@ async def test_empty_pkey_returns_empty_list(self, mock_ufm_config):
...
+            with pytest.raises(ApplicationError, match="unexpected response"):
                 await fetch_pkey_members(
                     FetchPKeyMembersInput(host="ufm.example.com", pkey="0x0005")
                 )
 
+    @pytest.mark.asyncio
+    async def test_pkey_404_reports_not_exists(self, mock_ufm_config):
+        """A 404 is tolerated: the partition is reported as missing and empty."""
+        with aioresponses() as m:
+            m.get(_pkey_members_url("0x0005"), status=404)
```
Comment (ryanheffernan) [inline src/tests/temporal/ngc/activities/test_ib_pkey_update.py:182]: worth adding a test to make sure the non-404 error case still raises?

Comment (susanhooks) [reply src/tests/temporal/ngc/activities/test_ib_pkey_update.py:182]: added

**Review by ryanheffernan — APPROVED** [review]
